dotcom-rendering/src/components/ShareButton.importable.tsx

```diff
@@ -184,13 +184,13 @@ export const ShareButton = ({
 		}),
 	};
 
-	// eslint-disable-next-line react-hooks/exhaustive-deps
 	useEffect(() => {
 		setIsShareSupported(
 			typeof navigator !== 'undefined' &&
 				'share' in navigator &&
 				navigator.canShare(shareData),
 		);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why did we disable this?

Reply:
[Actually, `navigator.canShare` does not take any argument](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/canShare). ESLint is correctly noticing that you are using a dependency that is unnecessary in your `useEffect` callback body.

Reply:
Actually you’re correct 🤦!

I would still keep the `shareData` in the dependency array. We should only optimise for reducing the number of renders if we believe there is a performance impact. Here I believe the value of `setIsShareSupported` will be set to the same value and therefor not cause subsequent renders.